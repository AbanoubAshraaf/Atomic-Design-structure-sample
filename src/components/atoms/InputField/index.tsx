import React from "react";

interface InputFieldProps {
  label: string;
  value: string | number;
  type: string;
  error?: string;
  onChange: (value: string) => void;
}

const InputField: React.FC<InputFieldProps> = ({ label, onChange, type, value, error }) => {

  return (
    <div className="flex flex-col gap-2 items-start">
      <label >{label}:</label>
      <input
        data-testid={label.toLocaleLowerCase()}
        onChange={(e) => onChange(e.target.value)}
        value={value}
        type={type}
        className="input input-bordered input-primary w-full max-w-xs"
      />
      {error && <span className="error-text">{error}</span>}
    </div>
  );
};

export default InputField;
