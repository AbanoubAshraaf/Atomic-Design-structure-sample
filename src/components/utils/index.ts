export const isValidAge = (age:number): boolean => {
    return (
      age > 0 && // Age must be positive
      !isNaN(age) && // Age must be a valid number
      Number.isInteger(age) // Age must be an integer
    );
  };
  