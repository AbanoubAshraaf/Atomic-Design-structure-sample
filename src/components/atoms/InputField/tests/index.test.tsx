import React from "react";
import { render, fireEvent } from "@testing-library/react";
import InputField from "..";
import { createSnapShot } from "@/components/test/utils";

describe("<InputField />", () => {
  test("render input field with label", () => {
    const { container, getByText } = render(
      <InputField
        label="Test Label"
        value=""
        type="text"
        onChange={() => {}}
      />,
    );
    expect(getByText("Test Label:")).toBeInTheDocument();
    createSnapShot(container);
  });

  test("display error message", () => {
    const { container, getByText } = render(
      <InputField
        label="Test Label"
        value=""
        type="text"
        error="Test Error"
        onChange={() => {}}
      />,
    );
    expect(getByText("Test Error")).toBeInTheDocument();
    createSnapShot(container);
  });

  test("call onChange handler", () => {
    const handleChange = jest.fn();
    const { getByTestId } = render(
      <InputField
        label="Test Label"
        value=""
        type="text"
        onChange={handleChange}
      />,
    );
    fireEvent.change(getByTestId("test label"), {
      target: { value: "new value" },
    });
    expect(handleChange).toHaveBeenCalledWith("new value");
  });
});
