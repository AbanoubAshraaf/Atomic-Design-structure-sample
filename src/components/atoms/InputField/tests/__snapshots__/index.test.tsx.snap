// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> display error message 1`] = `
<div>
  <div
    class="flex flex-col gap-2 items-start"
  >
    <label>
      Test Label
      :
    </label>
    <input
      class="input input-bordered input-primary w-full max-w-xs"
      data-testid="test label"
      type="text"
      value=""
    />
    <span
      class="error-text"
    >
      Test Error
    </span>
  </div>
</div>
`;

exports[`<InputField /> render input field with label 1`] = `
<div>
  <div
    class="flex flex-col gap-2 items-start"
  >
    <label>
      Test Label
      :
    </label>
    <input
      class="input input-bordered input-primary w-full max-w-xs"
      data-testid="test label"
      type="text"
      value=""
    />
  </div>
</div>
`;
